code/ship/ship.cpp

```diff
@@ -12454,6 +12466,10 @@ int ship_fire_secondary( object *obj, int allow_swarm, bool rollback_shot )
 				}
 			goto done_secondary;
 		}
+		if (shipp->flags[Ship_Flags::Secondaries_locked] && (obj == Player_obj && shipp->flags[Ship_Flags::Fail_sound_locked_secondary])) {
+			ship_maybe_do_secondary_fail_sound_hud(wip, false);
+			goto done_secondary;
+		}
```

Comment:
It would make more sense to do this above the above "due to weapons system damage" failure, because locked secondaries should take priority.